test/functionalTest/cases/2591_actionType_metadata_delete_support/actionType_metadata_delete_support.test

```diff
@@ -0,0 +1,188 @@
+#
+
+
+echo "01. Create entity E1 with attributes A:1, B:2, C:3"
+echo "=================================================="
+payload='{
+  "type": "T",
+  "id": "E1",
+  "A": {
+    "type": "Number",
+    "value": 1
+  },
+  "B": {
+    "type": "Number",
+    "value": 2
+  },
+  "C": {
+    "type": "Number",
+    "value": 3
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+
+echo "02. Subscribe to E.* for A, B and C; triggered by B, C"
+echo "======================================================"
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "idPattern": "E.*",
+        "type": "T"
+      }
+    ],
+    "condition": {
+      "attrs": [ "B", "C" ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [ "A", "B", "C" ],
+    "metadata": [ "previousValue", "actionType" ]
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+SUB_ID=$(echo "$_responseHeaders" | grep Location | awk -F/ '{ print $4 }' | tr -d "\r\n")
+
+
+
+echo "03. Delete attribute B"
+echo "======================"
+orionCurl --url /v2/entities/E1/attrs/B -X DELETE
+echo
+echo
+
+
+
+echo "04. Dump and reset: see notification with actionType=delete for B"
+echo "================================================================="
+accumulatorDump
+accumulatorReset
+echo
+echo
+
+
+
+--REGEXPECT--
+01. Create entity E1 with attributes A:1, B:2, C:3
+==================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E1?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Subscribe to E.* for A, B and C; triggered by B, C
+======================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. Delete attribute B
+======================
+HTTP/1.1 204 No Content
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+04. Dump and reset: see notification with actionType=delete for B
+=================================================================
+POST http://localhost:REGEX(\d+)/notify
+Fiware-Servicepath: /
+Content-Length: 260
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: normalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36}); cbnotif=1
+
+{
+    "data": [
+        {
+            "A": {
+                "metadata": {},
+                "type": "Number",
+                "value": 1
+            },
+            "B": {
+                "metadata": {
+                    "actionType": {
+                        "type": "Text",
+                        "value": "delete"
+                    }
+                },
+                "type": "Number",
+                "value": 2
```

Comment:
Currently, default behaviour is that deleted attribute doesn't appear in notifications (see delete_attribute_in_empty_condition.test and delete_attribute_in_condition.test test cases).

It could make sense to include deleted attributes in notifications when `actionType` is used (otherwise the `delete` case for `actionType` metadata is pointless). However, in this case, which should be used as `type` and `value` here?

1. The previous ones (as this PR currently shows)
2. Omit the `type` and `value` fields (but it should break the structure of attributes, as `value` is always assumed
3. Use null value, i.e. `"type": "None"` and `"value": null`

I think the most semantically correct is number 3.

Another interesting case should be using actionType and previousValue in combination. In this case, previousValue should show the value previous to the deletion. Something like this:

```
"B": {
                "metadata": {
                    "actionType": {
                        "type": "Text",
                        "value": "delete"
                    },
                    "previousValue": {
                        "type": "Number",
                        "value": 2
                    }
                },
                "type": "None",
                "value": null
```